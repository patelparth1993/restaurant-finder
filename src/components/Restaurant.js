import React, { Component } from "react";
import { connect } from "react-redux";

import { ViewRestaurant } from "./ViewRestaurant";
import { getRestaurants } from "../redux/actions/getRestaurant";
import "./style/Restaurant.scss";

class Restaurant extends Component {
  constructor(props) {
    super(props);
    this.state = {
      city: "",
      search: "",
      filterByPrice: 0,
      error: "",
    };
  }

  handleChange = (e) => {
    this.setState({
      [e.target.name]: e.target.value,
    });
  };

  getRestaurant = (e) => {
    if (e.target.className === "btn disabled") {
      this.setState({
        error: "Enter The City",
      });
    } else {
      this.setState({
        error: "",
      });
      this.props.getRestaurants(this.state.city);
    }
  };

  render() {
    const { city, search, filterByPrice, error } = this.state;
    return (
      <div className='container'>
        <div className='e1'>{error}</div>
        <div className='c1'>
          <input
            type='text'
            name='city'
            value={city}
            onChange={this.handleChange}
            placeholder='City'
          />
          <button
            onClick={this.getRestaurant}
            className={city ? "btn" : "btn disabled"}
          >
            Get Restaurant
          </button>
        </div>
        <div className='c2' hidden={this.props.restaurantsList.length === 0}>
          Refine:
          <input
            type='text'
            name='search'
            value={search}
            onChange={this.handleChange}
            placeholder='Name, Street Or Area'
          />
          <select
            className='select-filter'
            name='filterByPrice'
            value={filterByPrice}
            onChange={this.handleChange}
          >
            <option value='0' disabled hidden>
              By Price
            </option>
            <option value='1'>1</option>
            <option value='2'>2</option>
            <option value='3'>3</option>
            <option value='4'>4</option>
            <option value='0'>All</option>
          </select>
        </div>

        <ViewRestaurant
          resList={this.props.restaurantsList}
          search={search.toLowerCase()}
          filterByPrice={filterByPrice}
        />
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  restaurantsList: state.restaurants,
});
export default connect(mapStateToProps, { getRestaurants })(Restaurant);
