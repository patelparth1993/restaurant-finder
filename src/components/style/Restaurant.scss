.container {
  display: block;
  margin: auto;
  width: 100%;
  max-width: 1280px;

  input {
    padding: 6px;
    margin: 0px 3px;
    border-radius: 4px;
    border-color: none;
  }
  .e1 {
    text-align: center;
    color: red;
  }
  .c1 {
    text-align: center;
    margin-bottom: 10px;
    .btn {
      background-color: green;
      border: 0px;
      border-radius: 4px;
      padding: 8px;
      color: white;
      cursor: pointer;

      &.disabled {
        opacity: 0.6;
        cursor: unset;
      }
    }
  }
  .c2 {
    text-align: center;

    input {
      width: 9rem;
    }
    .select-filter {
      border-radius: 4px;
      padding: 5px;
    }
  }
}
