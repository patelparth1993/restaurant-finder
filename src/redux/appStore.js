import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";

import indexReducer from "./reducers/indexReducer";

export const appStore = applyMiddleware(thunk)(createStore)(
  indexReducer,
  window.devToolsExtension && window.devToolsExtension()
);
