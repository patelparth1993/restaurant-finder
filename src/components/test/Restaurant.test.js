import React from "react";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import { render, fireEvent, screen, cleanup } from "@testing-library/react";

import { appStore } from "../../redux/appStore";
import Restaurant from "../Restaurant";

afterEach(cleanup);

it("should take a snapshot", () => {
  const { asFragment } = render(
    <Provider store={appStore}>
      <Restaurant />
    </Provider>
  );

  expect(
    asFragment(
      <Provider store={appStore}>
        <Restaurant />
      </Provider>
    )
  ).toMatchSnapshot();
});

describe("Restaurant component renders", () => {
  it("disabled button", () => {
    const { getByTestId } = render(
      <Provider store={appStore}>
        <Restaurant />
      </Provider>
    );
    expect(getByTestId("get_restaurant_btn").className).toBe("btn disabled");
  });

  it("enabled button once city is entered", () => {
    const { getByTestId } = render(
      <Provider store={appStore}>
        <Restaurant />
      </Provider>
    );
    fireEvent.change(screen.getByLabelText("City"), {
      target: { value: "toronto" },
    });
    expect(getByTestId("get_restaurant_btn").className).toBe("btn");
  });
});
