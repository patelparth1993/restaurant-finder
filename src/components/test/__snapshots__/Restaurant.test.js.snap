// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="e1"
    />
    <div
      class="c1"
    >
      <label>
        City
        <input
          name="city"
          placeholder="Enter a City"
          type="text"
          value=""
        />
      </label>
      <button
        class="btn disabled"
        data-testid="get_restaurant_btn"
      >
        Get Restaurant
      </button>
    </div>
    <div
      class="c2"
      hidden=""
    >
      <label>
        Refine
        <input
          name="search"
          placeholder="Name, Street Or Area"
          type="text"
          value=""
        />
      </label>
      <select
        class="select-filter"
        name="filterByPrice"
      >
        <option
          disabled=""
          hidden=""
          value="0"
        >
          By Price
        </option>
        <option
          value="1"
        >
          1
        </option>
        <option
          value="2"
        >
          2
        </option>
        <option
          value="3"
        >
          3
        </option>
        <option
          value="4"
        >
          4
        </option>
        <option
          value="0"
        >
          All
        </option>
      </select>
    </div>
    <div
      class="view-restaurant-comp"
      data-testid="view-restaurant"
    />
  </div>
</DocumentFragment>
`;
